master/internal/task/allocation.go

```diff
@@ -961,6 +963,33 @@ func (a *allocation) exitedWithoutErr() bool {
 	return true
 }
 
+func (a *allocation) getErrCode() int {
+	for _, r := range a.resources.failed() {
+		code := r.Exited.Failure.ExitCode
+		if code != nil && *code != 0 {
+			return int(*code)
+		}
+	}
+	return 0
```

Comment:
I think we should make ```getErrCode()``` return a ```*int``` and return ```nil``` here

Reply:
If I return nil here, many instances where the allocation is supposed to exit with exit code 0 will exit with exit code null. Is that the desired behavior?

Reply:
Oh that is a good point, no we would want to record 0 if the containers report 0 

I was concerned with the case where it failed and we know it failed but we couldn't get the exit code, in Kubernetes this happens pretty frequently (that is honestly probably another bug)

Reply:
If aloocation.ExitErr is not nil, I am returning null if the getErrCode() function returns 0 so that should handle atleast some of those cases